public/app/features/browse-dashboards/components/BrowseActions/utils.ts

```diff
@@ -26,3 +39,269 @@ export function buildBreakdownString(
   }
   return breakdownString;
 }
+
+type MoveResourceMutations = {
+  createFile: ReturnType<typeof useCreateRepositoryFilesWithPathMutation>[0];
+  deleteFile: ReturnType<typeof useDeleteRepositoryFilesWithPathMutation>[0];
+};
+interface BulkMoveRequest {
+  selectedItems: Omit<DashboardTreeSelection, 'panel' | '$all'>;
+  targetFolderPath: string;
+  repository: RepositoryView;
+  mutations: MoveResourceMutations;
+  options: BaseProvisionedFormData;
+}
+
+interface MoveResultSuccess {
+  uid: string;
+  status: 'success';
+}
+
+interface MoveResultFailed {
+  uid: string;
+  status: 'failed';
+  title?: string;
+  failureType: 'create' | 'delete' | 'data-fetch';
+  error?: string;
+}
+
+export interface BulkMoveResult {
+  successful: MoveResultSuccess[];
+  failed: MoveResultFailed[];
+  summary: {
+    total: number;
+    successCount: number;
+    failedCount: number;
+  };
+}
+
+interface FulfilledResource {
+  uid: string;
+  data: ResourceWrapper;
+  title: string;
+}
+
+/**
+ * Execute bulk move operation for selected dashboards
+ * TODO: Follow up https://github.com/grafana/git-ui-sync-project/issues/315#issuecomment-3075920997 for better approach
+ */
+export const bulkMoveResources = async ({
```

Comment:
Should we consider naming this/use this as a hook instead? So it's `useBulkMoveResources`? As it feels like it's likely to be consumed in that manner, and then we would have rules-of-hooks eslint rules telling us off if necessary